### PR TITLE
Add support for system.stack_trace filtering optimizations for analyzer

### PR DESCRIPTION
**Pull request by azat**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add support for system.stack_trace filtering optimizations for analyzer

This should fix timeouts for `01051_system_stack_trace` test

Cc: @alexey-milovidov

### CODE REVIEW

File: src/Storages/System/StorageSystemStackTrace.cpp [hunk 1]
```diff
@@ -372,11 +382,55 @@ class StorageSystemStackTraceSource : public ISource
         }
 
         Block block { ColumnWithTypeAndName(std::move(all_thread_ids), std::make_shared<DataTypeUInt64>(), "thread_id") };
-        VirtualColumnUtils::filterBlockWithQuery(query, block, context);
+        VirtualColumnUtils::filterBlockWithPredicate(predicate, block, context);
```
[the remark this answers is not in the document]
Comment (azat) [reply src/Storages/System/StorageSystemStackTrace.cpp:385]: `filterBlockWithPredicate` has a check for `predicate == nullptr`

**Comment by azat** [comment]
>Stress test (msan) — Found signal in gdb.log

- https://github.com/ClickHouse/ClickHouse/issues/57508

>Stateless tests (tsan, s3 storage) [5/5] — fail: 1, passed: 1166, skipped: 17

- `00002_log_and_exception_messages_formatting` is flaky (this patch does not modifies any exceptions) (cc @tavplubix )

**Comment by tavplubix** [comment]
> 00002_log_and_exception_messages_formatting is flaky (this patch does not modifies any exceptions) (cc @tavplubix )

https://github.com/ClickHouse/ClickHouse/pull/58012

**Comment by azat** [comment]
@novikd Kind ping

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**